src/devices/Ahtxx/AhtBase.cs

```diff
@@ -0,0 +1,159 @@
+    /// Note: has been tested with AHT20 only, but should work with AHT10 and AHT15 as well.
+    /// Up to now all functions are contained in the base class, though there might be differences between the sensors types.
+    /// </summary>
+    public class AhtBase : IDisposable
+    {
+        /// <summary>
+        /// Address of AHTxx device (0x38). This address is fix and cannot be changed.
+        /// This implies that only one device can be attached to a single I2C bus at a time.
+        /// </summary>
+        public const int DeviceAddress = 0x38;
+
+        private enum StatusBit : byte // datasheet version 1.1, table 10
+        {
+            Calibrated = 0x08,
+            Busy = 0x80
+        }
+
+        private enum Command : byte
+        {
+            Calibrate = 0xbe,
+            SoftRest = 0xba,
+            Measure = 0xac
+        }
+
+        private I2cDevice _i2cDevice = null;
+        private double _temperature;
+        private double _humidity;
+
+        /// <summary>
+        /// Initializes a new instance of the device connected through I2C interface.
+        /// </summary>
+        /// <paramref name="i2cDevice">Reference to the initialized I2C interface device</paramref>
+        public AhtBase(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            SoftReset(); // even if not clearly stated in datasheet, start with a software reset to assure clear start conditions
+
+            // check whether the device indicates the need for a calibration cycle
+            // and perform calibration if indicated ==> c.f. datasheet, version 1.1, ch. 5.4
+            if (!IsCalibrated())
+            {
+                Calibrate();
+            }
+        }
+
+        /// <summary>
+        /// Gets the current temperature reading from the sensor.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Temperature GetTemperature()
+        {
+            Measure();
+            return new Temperature(_temperature, UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+        }
+
+        /// <summary>
+        /// Gets the current humidity reading from the sensor.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Ratio GetHumidity()
+        {
+            Measure();
+            return new Ratio(_humidity, UnitsNet.Units.RatioUnit.Percent);
+        }
+
+        /// <summary>
+        /// Perform sequence to retrieve current readings from device
+        /// </summary>
+        private void Measure()
+        {
+            Span<byte> buffer = stackalloc byte[3]
+            {
+                (byte)Command.Measure,
+                0x33, // command parameters c.f. datasheet, version 1.1, ch. 5.4
+                0x00
+            };
+            _i2cDevice.Write(buffer);
+
+            while (IsBusy())
+            {
+                Thread.Sleep(10);
```

Comment:
Are you sure you need so long loop times? What is the time it usually takes for this command to execute? (Probably specified in the sheet)

Reply:
Indeed I took the max. exec. times given in the datasheet. Probably I can shorten the cycle by querying I bit more often. I had this thought already, on the other hand speed wasn't my primary goal and at the same time I always try to avoid too much "polling" traffic on the bus. However, I think a 2ms interval wouldn't do any harm, but may speed up the whole reading process.

Reply:
I double checked the datasheet - it states: "Wait for 80ms to wait for the measurement to be completed. If the read status word Bit [7] is 0, it indicates that the measurement is completed, ...."
So looping in 10ms intervals should be good. 
However I did some testing and profiling. 

```
GetTemperature
     [1,2]: 0
     [2,5]: 0
    [5,10]: 0
   [10,20]: 983
   [20,50]: 14
  [50,100]: 2
 [100,200]: 0
```
It seems that the measurement takes much less than 80ms. So, I think the loop is ok.

Reply:
Thread.Sleep has resolution of about 12ms, IMO if this is expected time leave this as is, if this is much smaller then go with DelayHelper